core/src/main/java/hudson/model/Descriptor.java

```diff
@@ -814,21 +814,27 @@ public boolean configure( StaplerRequest req, JSONObject json ) throws FormExcep
             return configure(req);
         }
 
-        if (!Util.isOverridden(Descriptor.class, getClass(), "configure", StaplerRequest.class, JSONObject.class)) {
-            // Many descriptors used to override configure to implement databinding but also invoker super.configure().
-            // As we introduce default implementation this would result in duplicate binding + save() being called twice
-            // so, default "bindJSON" implementation is only used when configure isn't overriden.
-            try (BulkChange bc = new BulkChange(this)) {
-                req.bindJSON(this, json);
-                save();
-                bc.commit();
-            } catch (IOException e) {
-                LOGGER.log(Level.WARNING, "Failed to save "+getConfigFile(),e);
-            }
+        try (BulkChange bc = new BulkChange(this)) {
+            req.bindJSON(this, json);
+            bc.commit();
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Failed to save "+getConfigFile(),e);
         }
         return true;
     }
 
+
+    /**
+     * Configure descriptor instance form structured form submission where this specific descriptor is nested in
+     * a larger payload, which typically happens when Descriptor doesn't have a dedicated configuration view.
+     */
+    public boolean configureNested( StaplerRequest req, JSONObject json) throws FormException, IOException {
+        String name = getJsonSafeClassName();
+        JSONObject js = json.has(name) ? json.getJSONObject(name) : new JSONObject(); // if it doesn't have the property, the method returns invalid null object.
+        json.putAll(js);
```

Comment:
The change drops the inline comment explaining (mostly, it ends with an unspecified core version) why this madness seems needed.

Reply:
I felt it was more confusing than helping in understanding what this is all about - at least it didn't helped me. I'm still not sure why this is needed.

Reply:
```
// collapse the structure to remain backward compatible with the JSON structure before 1.
``
`
this comment was introduced by https://github.com/jenkinsci/jenkins/commit/4ab6fe7cb4ccf2277da970447c50eab7c0921e47  (jenkins-1.494) but still doesn't make any sense to me with this context information.

Nothing get "collapsed" here afaict and this method just assumes target descriptor's json is nested within a larger json payload, anything else ? Maybe this comment was copy/pasted from another source ?

Reply:
Well, without any comment you look at it and wonder why this mutates the passed object etc. Seems pretty insane. So restoring the extended comment and pointing to the commit perhaps would be useful?